Tools/MiniBrowser/wpe/main.cpp

```diff
@@ -336,6 +336,16 @@ static WebKitWebView* createWebViewForAutomationCallback(WebKitAutomationSession
     if (!useWPEPlatformAPI)
         return view;
 
+    // The original view might have been closed, so we need to find a valid view to clone
+    if (!g_hash_table_lookup(openViews, view)) {
+        GHashTableIter iter;
+        gpointer key, value;
+        g_hash_table_iter_init(&iter, openViews);
+        if (!g_hash_table_iter_next(&iter, &key, &value))
+            return nullptr;
+        view = WEBKIT_WEB_VIEW(value);
+    }
+
```

Comment:
I think we should do this check before the new api check, otherwise we would be returning a destroyed view for the old api. Maybe it would be better to add a way to track the active view instead of passing a view to the callback. But this solution is good for now, we can improve it in follow up patch.

Reply:
Good point about moving the check before returning the view. Indeed, in the case of the old API, while we always return the same original view _for automation_, the application might create new views from JS and continue to be alive even though the original view might have been deleted. In fact, we already register these new windows in the `openViews` hash table in the `createWebView` callback.

I'll move the new code before the old API check before merging.